BASE/src/Microsoft.ApplicationInsights/Channel/Transmission.cs

```diff
@@ -133,6 +162,18 @@ public ICollection<ITelemetry> TelemetryItems
             get; private set;
         }
 
+        /// <summary>
+        /// Send signal to Flush Async.
+        /// </summary>
+        public void SetFlushTaskCompletionSourceResult(bool status)
```

Comment:
Having looked at the usage, I think this is better expressed with three separate methods:
1. `CompleteFlush()`
2. `CancelFlush()`
3. `FailFlush(Exception ex)`